### PR TITLE
Minting UX

### PR DESCRIPTION
**Pull request by lykimq**
This PR contains:
- Add comments in the `index.ts` 
- Add progress bar for Mint
- Formatting code
- Add Readme
- Add Makefile
- Add ignore `*d.ts, *.js` in `.gitignore`
- Add progress bar for stashing

### CODE REVIEW

File: examples/nft/Makefile
Comment (quentin-burg) [inline examples/nft/Makefile:1]: Why do you need a Makefile ? 🧐 
You can just add some scripts into `package.json` and run `npm run myAwesomeCommand`

File: examples/nft/README.md [hunk 1]
```diff
@@ -1,5 +1,74 @@
...
+
+The web app allows you to mint a non-fungible token (NFT) and transfer it to a "staking" contract, even if you have no Tez in your wallet.
+
+
+## Prerequisites
+
+Before you start, ensure you have the following installed on your machine:
+
+- Node.js [Download and Install Node.js](https://nodejs.org/)
+- npm (Node Package Manager): Comes with Node.js installation
```
Comment (quentin-burg) [inline examples/nft/README.md:13]: You can talk about `nvm` (Node Version Manager) to manage easily Node versions : https://github.com/nvm-sh/nvm#installing-and-updating :-)

File: examples/nft/README.md [hunk 2]
```diff
@@ -1,5 +1,74 @@
...
+Navigate to the project folder:
+
+```
+cd examples/nft
+```
+
+To install dependencies, run:
+
+```
+make install
```
Comment (quentin-burg) [inline examples/nft/README.md]: or `npm i` 😉

File: examples/nft/src/routes/+page.svelte [hunk 3]
```diff
@@ -2,36 +2,44 @@
...
-  {/if}
-</div>
+  <p>
+    <empt>Minting</empt> a new NFT is achieved with a single call to the smart
+    contract. However, <empt>stashing</empt> an NFT to another smart contract requires
+    a transfer, which must be authorized off-chain through the signature of a permit.
+  </p>
+
+  <div>
+    {#if $myAccount == undefined}
```
Comment (quentin-burg) [inline examples/nft/src/routes/+page.svelte]: you need `===`  in Javascript, otherwise it doesn't test type

**Review by quentin-burg — APPROVED** [review]
Good ! Thanks 👍 😄

File: examples/nft/README.md (hunk 1 above)
Comment (lykimq) [reply examples/nft/README.md:13]: I update the nvm at this [commit](https://github.com/marigold-dev/gas-station-lib/pull/8/commits/ed9fb0e10837aa0546c666759690ed79081b7b10)

File: examples/nft/src/routes/+page.svelte (hunk 3 above)
Comment (lykimq) [reply examples/nft/src/routes/+page.svelte]: I add it at this [commit](https://github.com/marigold-dev/gas-station-lib/pull/8/commits/32cb02b77154073805717f22447157008b73eaef)

File: examples/nft/Makefile
Comment (lykimq) [reply examples/nft/Makefile:1]: I tend to forget the command run, that why :)

File: examples/nft/README.md (hunk 2 above)
Comment (lykimq) [reply examples/nft/README.md]: I prefer to keep it as `make install`, but yes it is equivalence. :)

File: index.ts [hunk 4]
```diff
@@ -4,15 +4,33 @@ import { packDataBytes } from "@taquito/michel-codec";
...
+ * Operation: Represents a generic blockchain operation with destination and parameters properties
+ */
+
 export type Operation = {
   destination: string;
   parameters: any;
 };
 
+/**
+ * TransferOperaiton: Represents a specific type of operation for transferring tokens,
```
Comment (aguillon) [inline index.ts]: `TransferOperaiton` is a typo

File: examples/nft/Makefile
Comment (aguillon) [reply examples/nft/Makefile:1]: I agree with @quentin-burg, please remove the Makefile, it is unecessary in this case.

**Review by aguillon — CHANGES_REQUESTED** [review]
Hey that's great, thank you! Can you just please remove the Makefile and use the `npm` related instructions in the README instead? And add the animation to the Stash button as well please.

Also, please remove the `text-align: center;` that I added for the `main` component? I don't know what I was thinking but yikes.

File: examples/nft/Makefile
Comment (lykimq) [reply examples/nft/Makefile:1]: I have drop the commit of the Makefile and change the README as well at this [commit](https://github.com/marigold-dev/gas-station-lib/pull/8/commits/0a7ed1f260fe991c423dfcc4dd5d480ff8829c3d)

File: index.ts (hunk 4 above)
Comment (lykimq) [reply index.ts]: Thanks, correct it at this [commit](https://github.com/marigold-dev/gas-station-lib/pull/8/commits/3dff2581e3e14ed88efb2883984400bdcfb03284)

Comment (lykimq) [reply index.ts]: I remove the `text-align:center` at this [commit](https://github.com/marigold-dev/gas-station-lib/pull/8/commits/53d47f800b92b1cb0490be0281422687bafd1dde#diff-d636b144ee28938d7218a83279a6f4ea1bc7ba85f567f9269bf0841d7a0cb169L41)

**Comment by quentin-burg** [comment]
Seems good for me !

**Comment by lykimq** [comment]
I add the stashing progress bar at this [commit](https://github.com/marigold-dev/gas-station-lib/pull/8/commits/25648eeac8ee774ec8ab43b6c7d1d39ebf26121b)

**Comment by aguillon** [comment]
Looks good thanks, let's test on staging.

**Pull request merged**